docs/brillig/01_architecture.md

```diff
@@ -0,0 +1,26 @@
+# Brillig Architecture
+
+
+Finite Field VM
+---
+The Brillig VM operates on finite fields using a 128 bit integer representation, matching the native representation in Noir and its backends. This decision simplifies SNARK proving and optimizes for efficiency. The exact maximum integer value the field can store should not be relied upon, however, it is assured to be capable of packing 3 32-bit integers. When a BinaryIntOp is used, the field is first cast to a fixed bitsize integer that can be accommodated within the field, prior to performing operations. Certain operations, such as ordered comparison or signed division, only make sense in the context of BinaryIntOp.
+
+Bytecode Structure and Function
+---
+The bytecode, an alternate compilation target of Noir code, is explored in more detail in another section of this document. The bytecode is available both standalone and as part of ACIR in the form of the Brillig opcode. It primarily operates on fixed register indices but also includes dedicated memory access operations. Designed as a minimal register machine, it supports conditional jumps, a lightweight callstack, and can access a flat memory array of field element cells, thereby fulfilling the core requirements of a low-level VM.
```

Comment:
"Brillig bytecode acts as an alternate compilation target for a domain specific circuit language. It will be explored in more detail in another section of this document."